src/pybind/mgr/rbd_support/schedule.py

```diff
@@ -318,7 +318,11 @@ def add(self,
     def remove(self,
                interval: Interval,
                start_time: Optional[StartTime] = None) -> None:
-        self.items.discard((interval, start_time))
+        if (interval, start_time) in self.items:
+            self.items.remove((interval, start_time))
+        else:
+            return "Interval does not exist({})".format(interval.to_string())
+        return ""
```

Comment:
I don't like returning an error/warning message this way. It looks weird.

I would just return a boolean (`false` if not found) and make the response message on a higher level.

Alternatively, having a tuple `(success(bool), response(str))` or `(code(int), response(ste))` instead would look a little better.

Reply:
Also, it isn't necessarily the interval that does not exist since a pair of `interval` and `start_time` is being looked up in the set.  Instead of attempting to differentiate between overall "schedule" not being there and particular settings not matching, I would just stick to generic "rbd: schedule already exists" and "rbd: schedule does not exist" messages.